src/libostree/ostree-repo-commit.c

```diff
@@ -2769,6 +2769,50 @@ ostree_repo_write_content (OstreeRepo       *self,
                                cancellable, error);
 }
 
+/**
+ * ostree_repo_write_regfile_inline:
+ * @self: repo
+ * @expected_checksum: (allow-none): The expected checksum
+ * @uid: User id
+ * @gid: Group id
+ * @mode: File mode
+ * @xattrs: (allow-none): Extended attributes, GVariant of type (ayay)
+ * @buf: (array length=len) (element-type guint8): File contents
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Synchronously create a file object from the provided content.  This API
+ * is intended for small files where it is reasonable to buffer the entire
+ * content in memory.
+ *
+ * Unlike `ostree_repo_write_content()`, if @expected_checksum is provided,
+ * this function will not check for the presence of the object beforehand.
+ *
+ * Returns: (transfer full): Checksum (as a hex string) of the committed file
+ */
+_OSTREE_PUBLIC
+char *
```

Comment:
Out of curiosity, isn't the convention for fallible functions to return a boolean?

Reply:
A pointer value being `NULL` signals an error by GLib convention - and gobject-introspection codifies this.  We have several other functions like this.

The problem then is APIs like this https://gitlab.gnome.org/GNOME/glib/-/commit/9966fe4493455dcdfe64483a50676891a878c72b